src/user/controller/password.ts

```diff
@@ -0,0 +1,46 @@
+import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import * as UserService from '../service';
+import { Forbidden, Unauthorized, NotFound } from '@curveball/http-errors';
+import { User } from '../types';
+import * as privilegeService from '../../privilege/service';
+
+class UserPasswordController extends Controller {
+
+  async put(ctx: Context) {
+
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege may create new users');
+    }
+
+    const userBody: any = ctx.request.body;
+
+    let user: User;
+    try {
+      user = await UserService.findByIdentity('mailto:' + userBody.identity) as User;
+    } catch (err) {
+      if (err instanceof NotFound) {
+        throw new NotFound('User email doesn\'t exist');
+      } else {
+        throw err;
+      }
+    }
+
+    const userNewPassword = userBody.newPassword;
+    const confirmNewPassword = userBody.confirmNewPassword;
+
+
+    if (userNewPassword !== confirmNewPassword) {
+      throw new Unauthorized('New password mismatch, please try again');
+    }
+
+    await UserService.updatePassword(user, userNewPassword);
+
+    ctx.response.status = 201;
+    ctx.response.headers.set('Location', `/user/${user.id}`);
```

Comment:
Can you explain the purpose of these 2 lines?

Reply:
i thought what this does is after `updatePassword` successfully passes, the status code will be 400 -> 201, then redirect you to `/user/:userId`?

Reply:
No, `201` is specifically for creating new resources, and when there is a Location header with a `201` it tells the client : We just created this resource at this uri.

So the way this is done right now, after changing a password the API says "I just created a user".

Reply:
So `Location` has 2 uses:

1. Redirects (all 3xx codes)
2. To tell a client where a new resource got created (only 201)